### PR TITLE
add a rolar log factory

### PR DESCRIPTION
**Pull request by jessitron**
this changes the type of LogFactory **Breaking** (hypothetically, no actual impact expected)
to accept (context, sdmGoal) instead of a lonely (name: string).

### CODE REVIEW

**Comment by jessitron** [comment]
@claymccoy  if you have this tested then merge it and then do sample?

File: src/handlers/events/delivery/goals/executeGoal.ts [hunk 1]
```diff
@@ -50,7 +50,7 @@ export async function executeGoal(rules: { projectLoader: ProjectLoader },
     const implementationName = sdmGoal.fulfillment.name;
     logger.info(`Running ${sdmGoal.name}. Triggered by ${sdmGoal.state} status: ${sdmGoal.externalKey}: ${sdmGoal.description}`);
 
-    await markGoalInProcess(ctx, sdmGoal, goal);
+    await markGoalInProcess({ctx: ctx, sdmGoal: sdmGoal, goal: goal, progressLogUrl: progressLog.url});
```
Comment (johnsonr) [inline src/handlers/events/delivery/goals/executeGoal.ts]: Use shorthand for `ctx` and `goal`

File: src/handlers/events/delivery/goals/executeGoal.ts [hunk 2]
```diff
@@ -161,20 +161,23 @@ function goalToHookFile(sdmGoal: SdmGoal, prefix: string): string {
...
 
-export function markStatus(ctx: HandlerContext, sdmGoal: SdmGoal, goal: Goal, result: ExecuteGoalResult, error?: Error) {
+export function markStatus(parameters: { ctx: HandlerContext, sdmGoal: SdmGoal, goal: Goal, result: ExecuteGoalResult, error?: Error, progressLogUrl: string }) {
+    let {ctx, sdmGoal, goal, result, error, progressLogUrl} = parameters;
     const newState = result.code !== 0 ? "failure" :
         result.requireApproval ? "waiting_for_approval" : "success";
     return updateGoal(ctx, sdmGoal,
         {
-            url: result.targetUrl,
+            url: result.targetUrl || progressLogUrl, // should we send both?
```
Comment (johnsonr) [inline src/handlers/events/delivery/goals/executeGoal.ts]: What's the difference. Is this documented anywhere?

File: src/spi/log/ProgressLog.ts [hunk 3]
```diff
@@ -23,7 +26,8 @@
 export interface ProgressLog {
 
     /**
-     * Name. Should relate to the immediate activity we're logging.
+     * Name. Should relate to the immediate activity wegs
```
Comment (johnsonr) [inline src/spi/log/ProgressLog.ts]: Fix typo

File: src/spi/log/ProgressLog.ts [hunk 4]
```diff
@@ -51,4 +55,4 @@ export interface ProgressLog {
     isAvailable(): Promise<boolean>;
 }
 
-export type LogFactory = (name: string) => Promise<ProgressLog>;
+export type LogFactory = (context: HandlerContext, sdmGoal: SdmGoal) => Promise<ProgressLog>;
```
Comment (johnsonr) [inline src/spi/log/ProgressLog.ts]: Makes sense

File: src/handlers/events/delivery/goals/executeGoal.ts (hunk 1 above)
Comment (jessitron) [reply src/handlers/events/delivery/goals/executeGoal.ts]: do you mean, don't use shorthand?

Comment (jessitron) [reply src/handlers/events/delivery/goals/executeGoal.ts]: ... oh you mean change it to say "ctx, goal"
This was a refactor, intellij does it this way

**Pull request merged**